scripts/build/builders/host.py

```diff
@@ -45,21 +46,71 @@ def BinaryName(self):
             raise Exception('Unknown app type: %r' % self)
 
 
-def ConcretePlatformName():
-    uname_result = uname()
-    return '-'.join([uname_result.system.lower(), release(), uname_result.machine])
+class HostBoard(Enum):
+    NATIVE = auto()
```

Comment:
This should likely just name x64.

The problem is that if your host is ARM64 these two boards are the same, so when you go to build "all boards" you build the same thing twice.

I don't now how to handle --board host, maybe an alias?

Reply:
It is an awkwardness because both 'platform host' and 'board native' mean the same thing. All other platforms make a tuple, this one stutters as the same info is duplicated.

Maybe placing the cross compile into platform was wrong ... will try to rethink.

Reply:
Updated logic a bit, now everything looks like `linux-x64-chip-tool` or `linux-arm64-all-clusters-app`.

Reply:
Still wondering what happens if you build 'all boards' on the Pi.

Reply:
I can try it out, but then arm64 will be filtered out and only the 'native' one will take over due to the condition:

```py
if HostBoard.NATIVE.BoardName() != HostBoard.ARM64.BoardName():
    _MATCHERS[Platform.HOST].AcceptBoard(Board.ARM64, board=HostBoard.ARM64)
```

The RPI will not accept the arm64 board, only native and native will be converted to arm64.

Reply:
I think it is a bug to only accept native. We should be able to paste command lines between host platforms and get the same results.

Reply:
I would prefer to do that as a followup. Native was a convenient way to say 'build whatever is appropriate to run chip-tool locally'

This can be done in defaults for sure, however it seems to move the intent of this PR (I want cross compile) a bit down the rabbit hole.

Reply:
Probably aliasing should be done, so that both 'arm64' and 'native' mean the same thing. This would mean I would also need to add x64 at least (don't think we have/need x86 currently) and logic for removing aliasing issues.

Reply:
I am not yet in favor of dropping native/host. It seems convenient to tell people to run `build_examples.py --platform host --app chip-tool` to build chip tool and not worry about what platform they are on.

Reply:
Tested on RPi (took a long time to bootstrap). Current PR version does what I would expect

```
ubuntu@ubuntu:~/connectedhomeip$ ./scripts/build/build_examples.py --platform host gen
2021-09-23 16:10:22 INFO    Platforms being built: host
2021-09-23 16:10:22 INFO    Boards being built: native
2021-09-23 16:10:22 INFO    Applications being built: all-clusters, chip-tool, thermostat
2021-09-23 16:10:22 INFO    Generating /home/ubuntu/connectedhomeip/out/linux-arm64-all_clusters
2021-09-23 16:10:22 INFO    Generating linux-arm64-all_clusters
2021-09-23 16:10:23 INFO    Done. Made 82 targets from 92 files in 792ms
2021-09-23 16:10:23 INFO    Command ['gn', 'gen', '--check', '--fail-on-unused-args', '--root=/home/ubuntu/connectedhomeip/examples/all-clusters-app/linux', '/home/ubuntu/connectedhomeip/out/linux-arm64-all_clusters'] completed
2021-09-23 16:10:23 INFO    Generating /home/ubuntu/connectedhomeip/out/linux-arm64-chip_tool
2021-09-23 16:10:23 INFO    Generating linux-arm64-chip_tool
2021-09-23 16:10:23 INFO    Done. Made 76 targets from 88 files in 770ms
2021-09-23 16:10:23 INFO    Command ['gn', 'gen', '--check', '--fail-on-unused-args', '--root=/home/ubuntu/connectedhomeip/examples/chip-tool', '/home/ubuntu/connectedhomeip/out/linux-arm64-chip_tool'] completed
2021-09-23 16:10:23 INFO    Generating /home/ubuntu/connectedhomeip/out/linux-arm64-thermostat
2021-09-23 16:10:23 INFO    Generating linux-arm64-thermostat
ls o2021-09-23 16:10:24 INFO    Done. Made 82 targets from 92 files in 776ms
2021-09-23 16:10:24 INFO    Command ['gn', 'gen', '--check', '--fail-on-unused-args', '--root=/home/ubuntu/connectedhomeip/examples/thermostat/linux', '/home/ubuntu/connectedhomeip/out/linux-arm64-thermostat'] completed
uubuntu@ubuntu:~/connectedhomeip$ ls out
linux-arm64-all_clusters  linux-arm64-chip_tool  linux-arm64-thermostat
```

Reply:
Also see pains in the way we do unit testing: we generate a 'dry run' and output commands.

But now we start having a difference between what is available in one platform vs others, so our unit test will fail on all arm64 platforms. I am not happy with where this is going, would rather look at it in following changes.